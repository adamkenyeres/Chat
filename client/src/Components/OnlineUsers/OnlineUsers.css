ul li::before {
    color: red;
}