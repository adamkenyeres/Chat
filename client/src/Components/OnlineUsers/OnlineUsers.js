import React, { Component } from 'react';
import './OnlineUsers.css'

class OnlineUsers extends Component {
    render() {
        const { users } = this.props;
        return (
            <div>
                <h3>The following users are online:</h3>
                <ul className="OnlineUsers-list">
                    {users.map(u => this.renderOnlineUsers(u))}
                </ul>

            </div>
        );
    }
    
    renderOnlineUsers(user) {
        return (
            <li className="OnlineUser">
                <div className="OnlineUser-content">
                    <div className="OnlineUser-username">
                        {user}
                    </div>
                </div>
            </li>
        );
    }
}

export default OnlineUsers;
