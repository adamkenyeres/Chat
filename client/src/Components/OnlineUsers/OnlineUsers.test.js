import React from 'react';
import { shallow } from 'enzyme';
import OnlineUsers from './OnlineUsers';


describe('Online Users', () => {
    it('should render correctly with no user', () => {
        const users = [''];
        const component = shallow(<OnlineUsers users={users} />);

        expect(component).toMatchSnapshot();
    });
    
    it('should render correctly with one user', () => {
        const users = ['Adam'];
        const component = shallow(<OnlineUsers users={users} />);

        expect(component).toMatchSnapshot();
    });

    it('should render correctly with multiple users', () => {
        const users = ['Adam', 'John', 'Josh'];
        const component = shallow(<OnlineUsers users={users} />);

        expect(component).toMatchSnapshot();
    });
});