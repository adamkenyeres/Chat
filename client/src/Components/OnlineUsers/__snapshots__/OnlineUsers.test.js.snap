// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Online Users should render correctly with multiple users 1`] = `
<div>
  <h3>
    The following users are online:
  </h3>
  <ul
    className="OnlineUsers-list"
  >
    <li
      className="OnlineUser"
    >
      <div
        className="OnlineUser-content"
      >
        <div
          className="OnlineUser-username"
        >
          Adam
        </div>
      </div>
    </li>
    <li
      className="OnlineUser"
    >
      <div
        className="OnlineUser-content"
      >
        <div
          className="OnlineUser-username"
        >
          John
        </div>
      </div>
    </li>
    <li
      className="OnlineUser"
    >
      <div
        className="OnlineUser-content"
      >
        <div
          className="OnlineUser-username"
        >
          Josh
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`Online Users should render correctly with no user 1`] = `
<div>
  <h3>
    The following users are online:
  </h3>
  <ul
    className="OnlineUsers-list"
  >
    <li
      className="OnlineUser"
    >
      <div
        className="OnlineUser-content"
      >
        <div
          className="OnlineUser-username"
        />
      </div>
    </li>
  </ul>
</div>
`;

exports[`Online Users should render correctly with one user 1`] = `
<div>
  <h3>
    The following users are online:
  </h3>
  <ul
    className="OnlineUsers-list"
  >
    <li
      className="OnlineUser"
    >
      <div
        className="OnlineUser-content"
      >
        <div
          className="OnlineUser-username"
        >
          Adam
        </div>
      </div>
    </li>
  </ul>
</div>
`;
